### PR TITLE
A bit of code review

### PR DESCRIPTION
**Pull request by AlCalzone**
Integration test is failing because of exit code 11. Shouldn't that be allowed (terminate)?

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -62,9 +60,9 @@ class Covid19 extends utils.Adapter {
 				await this.addUserCountriesTranslator();
 
 				// Write all country states depending on filter
-				for (const i in values) {
-					if (values.hasOwnProperty(i) && values[i] && values[i].country) {
-						let country = values[i].country;
+				for (const dataset of values) {
+					if (dataset.country) {
+						let country = dataset.country;
```
Comment (AlCalzone) [inline main.js]: values is an array. By iterating with `of` you can be sure that all values exist. Also I like speaking names, so `values[i] ==> dataset`

File: main.js [hunk 2]
```diff
@@ -73,51 +71,53 @@ class Covid19 extends utils.Adapter {
...
-										await this.localCreateState(country + '.flag', 'flag', values[i][y].flag);
-									}
-								} else if (this.config.loadAllCountrys === true) {
-									if (y !== 'countryInfo') {
-										await this.localCreateState(country + '.' + y, y, values[i][y]);
-									} else {
-										// Only take the flag from country info
-										await this.localCreateState(country + '.flag', 'flag', values[i][y].flag);
-									}
+						for (const property of Object.keys(dataset)) {
```
Comment (AlCalzone) [inline main.js]: Using `Object.keys` avoids checking `hasOwnProperty`

File: main.js [hunk 3]
```diff
@@ -73,51 +71,53 @@ class Covid19 extends utils.Adapter {
...
+							// Don't create a state for the country
+							if (property === 'country') continue;
```
Comment (AlCalzone) [inline main.js]: Exit condition instead of another indentation level.

File: main.js [hunk 4]
```diff
@@ -73,51 +71,53 @@ class Covid19 extends utils.Adapter {
...
+							if (
+								(!this.config.countries.length || this.config.countries.includes(dataset.country))
+								&& this.config.loadAllCountrys === false
```
Comment (AlCalzone) [inline main.js]: Just a bit of formatting for readability

File: main.js [hunk 5]
```diff
@@ -126,44 +126,42 @@ class Covid19 extends utils.Adapter {
...
-								await this.localCreateState('country_Top_5.' + count + '.' + y, y, values[i][y]);
-							} else {
-								// Only take the flag from country info
-								await this.localCreateState('country_Top_5.' + count + '.flag', 'flag', values[i][y].flag);
-							}
+				for (let position = 1; position <= 5; position++) {
+					const dataset = values[position - 1]; // start at 0
+					let country = dataset.country;
+
+					const channelName = 'country_Top_5.' + position;
```
Comment (AlCalzone) [inline main.js]: I'm not sure what you were trying to do here. Just use a normal for-loop from 1 to 5 and read the original array :)

File: main.js [hunk 6]
```diff
@@ -126,44 +126,42 @@ class Covid19 extends utils.Adapter {
...
+
+					await this.extendObjectAsync(channelName, {
```
Comment (AlCalzone) [inline main.js]: You're using the channel name a couple of times, so make it a const

File: main.js [hunk 7]
```diff
@@ -126,44 +126,42 @@ class Covid19 extends utils.Adapter {
...
+							await this.localCreateState(`${channelName}.flag`, 'flag', dataset[property].flag);
 						}
 					}
-					count = count + 1;
 				}
 
 
 				// Write continent information
-				if (continentsStats && this.config.getContinents === true) {
+				if (this.config.getContinents === true) {
```
Comment (AlCalzone) [inline main.js]: You initalize continentsStats at the start of the loop, so no need for that check.

File: main.js [hunk 8]
```diff
@@ -188,39 +186,38 @@ class Covid19 extends utils.Adapter {
...
 				const values = JSON.parse(result);
 
-				for (const i of Object.keys(values.features)) {
-					if (values.features[i]) {
-						this.log.debug('Getting data for Federal State : ' + JSON.stringify(values.features[i].attributes.LAN_ew_GEN));
-						const federalStateName = values.features[i].attributes.LAN_ew_GEN;
+				for (const feature of values.features) {
+					this.log.debug('Getting data for Federal State : ' + JSON.stringify(feature.attributes.LAN_ew_GEN));
+					const federalStateName = feature.attributes.LAN_ew_GEN;
+					const channelName = `Germany.Federal_States.${federalStateName}`;
```
Comment (AlCalzone) [inline main.js]: values.features is an array. Just iterate it directly.

File: main.js [hunk 9]
```diff
@@ -75,22 +75,16 @@ class Covid19 extends utils.Adapter {
...
+								!this.config.countries.length
+								|| this.config.countries.includes(dataset.country)
+								|| this.config.loadAllCountrys
 							) {
 								if (property !== 'countryInfo') {
 									await this.localCreateState(country + '.' + property, property, dataset[property]);
 								} else {
 									// Only take the flag from country info
 									await this.localCreateState(country + '.flag', 'flag', dataset[property].flag);
 								}
```
Comment (AlCalzone) [inline main.js]: Those two code blocks were duplicated, so I combined the logic.

File: main.js (hunk 1 above)
Comment (DutchmanNL) [reply main.js]: nice one thanks !

**Pull request merged**